### ISSUE TITLE
net.createServer still uses real file system

### ISSUE THREAD

**Issue by TheDirigible** [tschaub/mock-fs#370]
When using net.createServer to create a unix domain socket (socket file), it is created in the real file system, even though mock-fs was used.  Is there any fix for this?

**Comment by 3cp** [comment]
mock-fs is not an implementation of file system, it only works in nodejs context by monkey patching some methods. Socket file is probably out of reach.

For your use case, you can try to use a real file system like [tmpfs](https://www.kernel.org/doc/html/latest/filesystems/tmpfs.html) to create a transient file system in ram.

**Comment by 3cp** [comment]
Yes, besides normal file and directory, mock-fs only has symlink.
